### PR TITLE
Added hikaricp.connections.max (pool max size)

### PR DESCRIPTION
**Pull request by styner9**
### What does this PR do?
Add `hikaricp.connections.max` for HikariCP max connection pool size.

### Motivation
I wnat to calculate a pool utilization. To do so, I need to know the maximum size and the number of idle/busy connections.
But, max size is not provided.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes
N/A

### CODE REVIEW

File: hikaricp/metadata.csv
Comment (kayayarai) [inline hikaricp/metadata.csv:6]:
```suggestion
hikaricp.connections.max,gauge,,,,Max number of connections,0,hikaricp,max connection number,
```
Maybe?

**Review by kayayarai — CHANGES_REQUESTED** [review]

File: hikaricp/metadata.csv
Comment (styner9) [reply hikaricp/metadata.csv]: @kayayarai Thanks!

**Review by kayayarai — APPROVED** [review]

**Comment by styner9** [comment]
@kayayarai Hi! I can't merge this PR because I don't have write access. Can you merge it for me?

File: hikaricp/metadata.csv
Comment (FlorentClarret) [inline hikaricp/metadata.csv:6]: nit, we have a unit `connection`

```suggestion
hikaricp.connections.max,gauge,,connection,,Max number of connections,0,hikaricp,max connection number,
```

**Review by FlorentClarret — CHANGES_REQUESTED** [review]
Thanks for the PR! 

Just left one comment on the metadata file

File: hikaricp/metadata.csv
Comment (styner9) [reply hikaricp/metadata.csv]: @FlorentClarret Thank you!

**Review by FlorentClarret — APPROVED** [review]
👍

**Comment by FlorentClarret** [comment]
Pinging @bertaudamien as the maintainer of this integration

**Comment by styner9** [comment]
@bertaudamien Hi! Could you review/merge this PR? Thank you in advance.

**Comment by FlorentClarret** [comment]
Merging. I'll release a new version right after

**Pull request merged**
